src/pidigits.rs

```diff
@@ -27,13 +15,10 @@ fn main() {
         .and_then(|s| s.into_string().ok())
         .and_then(|n| n.parse().ok())
         .unwrap_or(27);
-    for (i, d) in Context::new().enumerate().take(n) {
-        print!("{}", d);
-        if (i + 1) % 10 == 0 { println!("\t:{}", i + 1); }
-    }
-    if n % 10 != 0 {
-        for _ in n % 10 .. 10 { print!(" "); }
-        println!("\t:{}", n);
+    let digits: Vec<_> = Context::new().take(n).collect();
+    for (i, chunk) in digits.chunks(10).enumerate() {
+        let n = chunk.iter().fold(0, |n, d| n * 10 + *d as u64);
+        println!("{:<10}\t:{}", n, i * 10 + chunk.len());
```

Comment:
I don't thing it's working: if the numbers of a line are '00236401', you'll print '236401'. And I think it's better to print the digits, that's easier to understand.